### PR TITLE
Partner pca templates new pdf service

### PR DESCRIPTION
**Pull request by csabadenes**
Move PDF generation to a separate component

### CODE REVIEW

**Comment by csabadenes** [comment]
Relates to:
https://github.com/unicef/etools-infra/pull/39
https://github.com/unicef/etools-nginx-proxy/pull/6
https://github.com/unicef/etools-pdf-service

This PR attempts to move the PCA PDF generation functionality to a new docker machine/service. 
The reason for this move is that one of the templates(arabic) requires RTL text support. To achieve this, i had installed some extra libraries, which were considered too heavy to be put on the actual Web server, 
so they were moved to a new machine/Service instead.

The current backend API for the PCA PDF generation should stay unchanged, the backend will make a background request to the PDF service, sending the required context, and the PDF machine should return the generated PDF document.

The authentication is done by sending the HTTP Basic Auth header with every request. The user/pass should be configured as an ENV variable on every server(local/dev/demo/staging/prod).

Another note: i had to add a new and restore few of the recently cleaned up serializers to "partners/serializers/v1.py".

File: EquiTrack/partners/serializers/v1.py [hunk 1]
```diff
@@ -10,3 +15,31 @@ class FileTypeSerializer(serializers.ModelSerializer):
     class Meta:
         model = FileType
         fields = '__all__'
+
+
+class PartnerStaffMemberSerializer(serializers.ModelSerializer):
```
Comment (ntrncic) [inline EquiTrack/partners/serializers/v1.py]: can we use any of the serializers in `partner_organization_v2`.
Maybe we should move the views and serializers to v2 files.

File: EquiTrack/EquiTrack/utils.py [hunk 2]
```diff
@@ -52,6 +53,24 @@ def get_data_from_insight(endpoint, data={}):
     return True, result
 
 
+def load_internal_pdf_template(endpoint='', data={}):
+    api_url = 'http://{}/{}/'.format(
+        settings.PDF_API_URL,
+        'api/pdf/' + endpoint
+    )
+
+    response = requests.post(
```
Comment (ntrncic) [inline EquiTrack/EquiTrack/utils.py]: why a post request?

File: EquiTrack/EquiTrack/settings/base.py [hunk 3]
```diff
@@ -540,6 +540,10 @@ def str2bool(value):
 VISION_USER = os.getenv('VISION_USER', 'invalid_vision_user')
 VISION_PASSWORD = os.getenv('VISION_PASSWORD', 'invalid_vision_password')
 
+PDF_API_URL = os.getenv('PDF_API_URL', '')
+PDF_API_USER = os.getenv('PDF_API_USER', '')
```
Comment (ntrncic) [inline EquiTrack/EquiTrack/settings/base.py]: I think we are working on JWT authentication. maybe we can send a token to the container. 
@robertavram what do you think?

Comment (csabadenes) [reply EquiTrack/EquiTrack/settings/base.py]: I can look at the current JWT auth implementation and try to port it to this service.

File: EquiTrack/EquiTrack/utils.py (hunk 2 above)
Comment (csabadenes) [reply EquiTrack/EquiTrack/utils.py]: POST request is required to send the context data(not possible with GET)

File: EquiTrack/partners/serializers/v1.py (hunk 1 above)
Comment (csabadenes) [reply EquiTrack/partners/serializers/v1.py]: Sure, i can refactor it to v2.

File: EquiTrack/EquiTrack/settings/base.py (hunk 3 above)
Comment (csabadenes) [reply EquiTrack/EquiTrack/settings/base.py]: I have also asked Rob about the auth, i think the conclusion was that JWT auth does not really worth to be implemented for this scenario. 
Also, JWT has multiple steps(fetching a token from the server), which would slow down the request, unless we will use a pre-generated token(not sure how?) which would be basically the same as the current auth, as far as i can tell.

File: EquiTrack/partners/serializers/v1.py (hunk 1 above)
Comment (csabadenes) [reply EquiTrack/partners/serializers/v1.py]: I have moved the serializers to v2 for both partners and reports, hopefully it is compatible with the direction of the latest related changes.

**Comment by ntrncic** [comment]
@csabadenes please resolve conflicts. sorry

**Comment by domdinicola** [comment]
please rebase the PR

**Comment by domdinicola** [comment]
@csabadenes can you resolve conflicts

**Pull request closed without merging**